doc/rst/technotes/libraries.rst

```diff
@@ -551,6 +558,149 @@ include the chapel runtime and standard modules for use in a non-Chapel program
+
+This example demonstrates explicitly adding the arguments in the program using
+the ``foo`` library.
+
+.. code-block:: C
+
+   #include "foo.h"
+
+   int main(int argc, char* argv[]) {
+     int argChapelC = 3;
+     char* argChapelV[3] = {argv[0], "-nl", "2"};
+     // Initialize the Chapel runtime and standard modules
+     chpl_library_init(argChapelC, argChapelV);
+
+     baz(7); // Call into a library function
+
+     chpl_library_finalize();
+
+     return 0;
+   }
+
+Alternatively, the original single locale client from `Initializing Your
+Library In C`_ can be used with the additional two arguments to the executable:
+
+.. code-block:: bash
+
+   ./a.out -nl 2
+
+Users also still need to call the generated module initialization function for
+multilocale libraries, as mentioned in that section.
+
+In Python
+~~~~~~~~~
+
+Users must still call ``chpl_setup()`` before utilizing the exported Chapel
+functions.  However, it requires a ``numLocales`` argument when the library
+has been compiled for multilocale settings.  E.g. to run with ``4`` locales,
+write ``chpl_setup(4)``.
+
+Makefile-less Compilation
+-------------------------
+
+When compiling a C program using a multilocale Chapel library without a
+makefile, some additional steps are needed beyond those required for using a
+:ref:`single locale Chapel library <Makefileless Compilation In Single Locale>`.
+For this reason, we strongly suggest using the Makefile generated by the
+``--library-makefile`` flag, as described :ref:`here <Makefile Helper>`.
+
+A new ``compileline`` tool, ``compileline --multilocale-lib-deps`` is currently
+required after ``compileline --libraries``.  Additionally, the library being
+linked must be listed twice in the compilation command - once before
+``compileline --libraries`` and once at the end of the command.  The compilation
+command would then look like this (replacing ``myCProg.c`` with the name of your
+C program that will use the library):
+
+.. code-block:: sh
+
+   `$CHPL_HOME/util/config/compileline --compile` myCProg.c -Llib/ -lfoo `$CHPL_HOME/util/config/compileline --libraries` `$CHPL_HOME/util/config/compileline --multilocale-lib-deps` -lfoo
+
+What is this _server_real program?
+----------------------------------
+
+When you compile a Chapel library for use with multiple locales, you should
+typically see both a library (see :ref:`Location of the Generated Library` for
+where this will be placed and how to control that location) and a binary (which
+is currently generated in the same directory as the "main" source file), in
+addition to other support files such as the generated header, makefile, etc.
+
+The library will be named as specified in :ref:`Library Name`; the binary will
+use the same base name as the library (omitting the ``lib`` and ``.so`` or
+``.a`` portions), followed by ``_server_real``.  Thus for a library
+``libfoo.so``, the binary would be named ``foo_server_real``.
+
```

Comment:
It'd be nice to have a section of self-contained code-block examples demonstrating how to compile and run a simple hello-world multilocale Chapel library and call it from python (for simplicity sake). This could speed up the process of someone trying to build get basic example up and running themselves. I am thinking of a structure like this:

```chpl
// Contents of hello.chpl
...
```

```python
# Contents of call_hello.py
...
```

```bash
# Command line steps to build the library and run it
...
```

This section might be a good place to insert that, or we could put it into a new "Example" section just after this.

What do you think?

Reply:
I think that sounds more along the lines of the primer I wanted to write for this feature